lighthouse-core/audits/installable-manifest.js

```diff
@@ -51,59 +125,73 @@ class InstallableManifest extends MultiCheckAudit {
   }
 
   /**
-   * @param {LH.Artifacts.ManifestValues} manifestValues
-   * @return {Array<string>}
+   * @param {LH.Artifacts} artifacts
+   * @return {Array<LH.IcuMessage>}
    */
-  static assessManifest(manifestValues) {
-    if (manifestValues.isParseFailure && manifestValues.parseFailureReason) {
-      return [manifestValues.parseFailureReason];
+  static getInstallabilityErrors(artifacts) {
+    const installabilityErrors = artifacts.InstallabilityErrors.errors;
+
+    const errorMessages = [];
+    for (const err of installabilityErrors) {
+      // @ts-expect-error errorIds from protocol should match up against the strings dict
+      const matchingString = UIStrings[err.errorId];
+      // We only expect a `minimum-icon-size-in-pixels` errorArg[0] for two errorIds, currently.
```

Comment:
this will throw if this statement becomes not true at some point in the future (new error type or whatever), so better to check this as well :)

Reply:
a brand new error id that our code doesn't handle will fall into L157.

an existing error id that our code handles could have an arg added to it, and we'd fall into L164 but the error message string wouldn't handle the values. i18n will throw. https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/lib/i18n/i18n.js#L267

we could avoid errors if `_str` somehow provided how many arguments a formatted string has, and check if that is equal to `errorArguments.length`, and if not we fallback to a generic string (`${errorId} ${errorArguments.join(', ')}`). But `_str` doesn't provide # of args. We could do a string search for `value#` for each index in `errorArguments` as a substitute ...

Reply:
@connorjclark What do you think about having something like this? 

`
const errorArgumentsRegex = /{([^}]+)}/g;
const UIStringArguments = matchingString.match(errorArgumentsRegex) || [];
if (matchingString && value0) {
        errorMessages.push(str_(matchingString, {value0}));
      } else if (matchingString && err.errorArguments.length !== UIStringArguments.length) {
        // Matching string, but have the incorrect number of arguments for the message.
        errorMessages.push(str_(UIStrings.notEnoughArguments, {errorId: err.errorId}))
      } else if (matchingString) {
        errorMessages.push(str_(matchingString));
      } else {
        errorMessages.push(str_(UIStrings.noErrorId));
      }
`
Where the `notEnoughArguments` string would be the generic string you mentioned. Although it doesn't include the args in the string. Somewhat similar to how [Devtools does it](https://source.chromium.org/chromium/chromium/src/+/master:third_party/devtools-frontend/src/front_end/resources/AppManifestView.js;l=332;drc=159aba9ef8e861eea84ce653fb14b8183420f279)

Reply:
looks good, that's a tricky regex. couple things

1) the first case is still susceptible to an error (if we get an unexpected number of args for an error we already handle).
2) the check for `matchingString && err.errorArguments.length !== UIStringArguments.length` should go first, that'd solve thing 1)
3) let's add a second arg to `UIStrings.notEnoughArguments`, so the string can attempt to be useful (at the risk of printing randomish data)

I suspect the "1 arg case" is fine forever, even if a "2 arg case" comes up still just duplicating a little code instead of making a generic "n arg case" is much better to read.

Reply:
shane says let's avoid UI strings/i18n for the "notEnoughArguments" case